connection.go

```diff
@@ -633,10 +627,23 @@ func (c *Connection) sendCombinedPacket(pt uint8, frames []frame, acks ackRanges
 	asent := int(0)
 	var err error
 
-	if len(acks) > 0 {
-		var af *frame
+	for _, f := range frames {
+		l, err := f.length()
+		if err != nil {
+			return 0, err
+		}
+		left -= l
+	}
 
-		af, asent, err = c.makeAckFrame(acks, left)
+	// See if there is space for any acks, and if there are acks waiting
+	maxackblocks := (left - 16) / 5 // We are using 32-byte values for all the variable-lengths
+	if maxackblocks > 255 {
+		maxackblocks = 255
+	}
+
+	if len(acks) > 0 && (left - 16) >= 0 {
```

Comment:
Instead of having maxackblocks computed here, we should compute it in makeAckFrame, because then we don't need to know how large it is.